src/coreclr/src/jit/liveness.cpp

```diff
@@ -2040,7 +2040,19 @@ void Compiler::fgComputeLifeLIR(VARSET_TP& life, BasicBlock* block, VARSET_VALAR
                     // (as opposed to side effects of their children).
                     // This default case should never include calls or assignments.
                     assert(!node->OperRequiresAsgFlag() && !node->OperIs(GT_CALL));
-                    if (!node->gtSetFlags() && !node->OperMayThrow(this))
+
+                    bool isR2RIndirectParam = false;
+#if defined(FEATURE_READYTORUN_COMPILER) && defined(TARGET_ARMARCH)
+                    if (node->OperGet() == GT_IND)
+                    {
+                        GenTree* addr = node->AsIndir()->Addr();
+                        if (addr->OperGet() == GT_PHYSREG)
+                        {
+                            isR2RIndirectParam = addr->AsPhysReg()->gtSrcReg == REG_R2R_INDIRECT_PARAM;
```

Comment:
It looks to me like this code will eliminate any `IND(PHYSREG x11)`. It's possibly the case that the only way these are getting introduced today is in your change to lower.cpp. But is it safe, long-term? Not every use of x11 is a R2RIndirectParam.

It seems like you need to delete only, and exactly, the cases where these are added as a special call argument. Wouldn't that be in the `GT_CALL` case above, where it's looking through the arguments of a call that it has already decided to remove?